src/buffer/out/textBuffer.cpp

```diff
@@ -3316,8 +3315,6 @@ MarkExtents TextBuffer::_scrollMarkExtentForRow(const til::CoordType rowOffset,
 
                     endThisMark(lastMarkedText.x, lastMarkedText.y);
                 }
-                // Otherwise, we've changed from any state -> any state, and it doesn't really matter.
-                lastMarkKind = markKind;

```

Comment:
BTW the `lastMarkKind` variable is entirely unused. The condition above, `if (markKind != MarkKind::None)` also seems wrong to me. Is it supposed to test `lastMarkKind`?